### ISSUE TITLE
This plugin keeps spamming my /tmp with 'tmp-<code>.tmp' files

### ISSUE THREAD

**Issue by andrevmatos** [mhartington/nvim-typescript#226]
# Problems summary

After a couple of seconds of opening a project with a few .ts files, my /tmp gets a lot of tmp files like `tmp-43380hjTyrVgkp5T.tmp`. These files lay around and aren't even deleted when exiting neovim.

## Expected

Create the files in memory or clean up after them (preferably creating them in a sub/tempfolder as well). Seems like the files have the content of my opened files, so I suppose this line is the culprit:
https://github.com/mhartington/nvim-typescript/blob/eca2bb92d45f09fc500317ededace5bc849063c1/rplugin/node/nvim_typescript/src/index.ts#L842

## Environment Information

 * terminal: Konsole/zsh
 * vim/nvim version: neovim v0.3.4

## Provide a minimal vim rc with less than 50 lines (Required!)

```vim
if &compatible
  set nocompatible               " Be iMproved
endif

" Required:
set runtimepath+=~/.cache/nvim_dein/repos/github.com/Shougo/dein.vim

if dein#load_state('~/.cache/nvim_dein')
  call dein#begin('~/.cache/nvim_dein')

  " Let dein manage dein
  " Required:
  call dein#add('~/.cache/nvim_dein/repos/github.com/Shougo/dein.vim')

  call dein#add('HerringtonDarkholme/yats.vim')
  call dein#add('mhartington/nvim-typescript', {'build': './install.sh'})
  call dein#add('Shougo/deoplete.nvim')
  call dein#add('Shougo/denite.nvim')
  " Required:
  call dein#end()
  call dein#save_state()
endif

autocmd FileType typescript nmap <Leader><space> :TSType<CR>
autocmd FileType typescript nmap <Leader>d :TSDef<CR>
autocmd FileType typescript nmap <Leader>r :TSRename<CR>
autocmd FileType typescript nmap <Leader>i :TSImport<CR>
```

## The reproduce ways from neovim starting

 1. Open neovim's `session.vim` with a couple of .ts files on it
 2. See the `/tmp/tmp-*.tmp` files be created

## Screen shot (if possible)

![image](https://user-images.githubusercontent.com/587021/54971823-12f85080-4f67-11e9-91b5-07947878c812.png)

**Comment by andrevmatos** [comment]
The fix didn't work. But now the files are all empty, but still left there even after I exit nvim. I verified that the line I pointed above is actually and still the culprit on current master by adding `{ prefix: 'nvim_ts-' }` option to `fileSync` and seeing the `nvim_ts-xxxxxxxx.tmp` files starting to spam my `/tmp`. That was the case even by changing `.then` to `.finally` to call the `removeCallback`.
Any way of using fd directly on an in-mem solution instead of actual files for the `client.updateFile` call? Maybe any solution with `stdin` pipe, as usual with vim syntax checkers?
